raster_analysis/query_executor.py

```diff
@@ -93,13 +104,22 @@ def _aggregate_window_by_group(
         elif aggregate.layer == SpecialSelectors.area__ha:
             return SpecialSelectors.area__ha, group_counts * self.data_cube.mean_area
         else:
+            layer = self.query.data_environment.get_layer(aggregate.layer)
             window = self.data_cube.windows[aggregate.layer]
             masked_data = np.extract(mask, window.data)
 
+            # is NoData is NaN, need to remove values from before aggregation
+            if np.isnan(layer.no_data):
```

Comment:
Don't NaN values get masked out like other `no_data` values in `mask` that's passed down to this function? Curious of why the special treatment is needed.

Reply:
@solomon-negusse that mask only masks out things like WHERE clause and GROUP BY filters. This part aggregates the remaining layers, in which case there might be still unmasked NaN values. 

The reason we can't mask this out earlier is that if we're aggregating across multiple float rasters at once, one may layer may have NaN values where the other doesn't, and these shouldn't affect each other.

E.g. see test_net_flux, which is how I found this issue. It gets the sum of 3 different float layers, filtered to a certain tree cover density. So the mask containers the tree cover filter, but each float layer needs to be independently summed.

Reply:
Thanks @jterry64, that was mainly for my edification and makes sense. The updates look LGTM.